### PR TITLE
Changes mekare

### PR DESCRIPTION
**Pull request by erakem**
# What's new

I tried to read the relevant documentation but not sure.

cli_cmd_interrupt_received: easier to read
cli_getc: easier to read
furi_hal_nfc_detect: code simplified a little bit
furi_hal_nfc_emulate_nfca: indent skip
furi_hal_nfc_listen_rx: data_received removed
furi_hal_nfc_transparent_tx_rx: remove useless ret variable
furi_hal_nfc_tx_rx_get_flag: defaulting to RFAL_TXRX_FLAGS_CRC_TX_AUTO (=0)
furi_string_cat_printf: fixed missing format
ibutton_worker_read_comparator: remove double break
lfrfid_app: factorize call to DOLPHIN_DEED
lfrfid_cli_raw_emulate: move overrun check before FURI_BIT's call
lfrfid_cli_raw_read: move overrun check before FURI_BIT's call
lfrfid_cli_write: slightly easier to read
lfrfid_load_key_data: simplify code
lfrfid_save_key_data: simplify code
nfc_emv_parser_get_aid_name: change parsed assignation
nfc_emv_parser_get_country_name: change parsed assignation
nfc_emv_parser_get_currency_name: change parsed assignation
protocol_dict_decoders_feed_by_feature: should work as long as ProtocolDecoderFeeds have no side effect
protocol_dict_decoders_feed_by_id: should work as long as ProtocolDecoderFeeds have no side effect
protocol_dict_decoders_feed: should work as long as ProtocolDecoderFeeds have no side effect
protocol_dict_encoder_start: simplified return
protocol_dict_render_brief_data: useless return?
protocol_dict_render_data: useless return?
rfalNfcGetActiveDevice: typo in doc
subghz_keystore_raw_encrypted_save: while condition is irrelevant
subghz_protocol_decoder_bett_feed: remove reassign same value

# Verification 

- I intended to not change anything and only rewrite some part of the code I looked at. But I might be wrong

# Checklist (For Reviewer)

- [ ] PR has description of feature/bug or link to Confluence/Jira task
- [ ] Description contains actions to verify feature/bugfix
- [ ] I've built this code, uploaded it to the device and verified feature/bugfix

### CODE REVIEW

File: lib/subghz/protocols/bett.c [hunk 1]
```diff
@@ -242,7 +242,6 @@ void subghz_protocol_decoder_bett_feed(void* context, bool level, uint32_t durat
         if(!level) {
             if(DURATION_DIFF(duration, subghz_protocol_bett_const.te_short * 44) <
                (subghz_protocol_bett_const.te_delta * 15)) {
-                instance->decoder.parser_step = BETTDecoderStepSaveDuration;
                 if(instance->decoder.decode_count_bit ==
```
Comment (Skorpionm) [inline lib/subghz/protocols/bett.c]: why did you remove this line? Have you seen how the protocol itself looks like? she is needed there

Comment (erakem) [reply lib/subghz/protocols/bett.c]: l232 switch(instance->decoder.parser_step) {
l241 case BETTDecoderStepSaveDuration:
l245 instance->decoder.parser_step = BETTDecoderStepSaveDuration;

so it is already set at the good value, no ?

File: applications/main/lfrfid/lfrfid.c [hunk 2]
```diff
@@ -249,31 +247,26 @@ bool lfrfid_delete_key(LfRfid* app) {
 }
 
 bool lfrfid_load_key_data(LfRfid* app, FuriString* path, bool show_dialog) {
-    bool result = false;
-
-    do {
-        app->protocol_id = lfrfid_dict_file_load(app->dict, furi_string_get_cstr(path));
```
Comment (skotopes) [inline applications/main/lfrfid/lfrfid.c:256]: Removing this causing crash

**Review by skotopes — CHANGES_REQUESTED** [review]

**Comment by skotopes** [comment]
Please verify all your changes again, ensure that you've covered all cases.

File: applications/main/lfrfid/lfrfid.c (hunk 2 above)
Comment (erakem) [reply applications/main/lfrfid/lfrfid.c:256]: Indeed I missed the update of app->protocol_id. My bad.

**Comment by skotopes** [comment]
@erakem have you also tested all apps and features touched by this PR?

Asking cause it's going to take a lot of efforts on our side and I want to make sure that you actually have tested all those changes that you've made.

**Comment by erakem** [comment]
It makes perfect sense.
I am doing another pass but I am unsure how to correctly test it. So I will review again my code too.

I am currently facing a bug when I try to : 
1) Read an RFID card
2) Write it back without saving it (so "More" > "Write")
3) Backlight goes off and stays off <--- ?!

It breaks too when I do this : Saved / <select one> / Write / Back / Write / Back / Write
I followed the bug up to notification_message

I thought it was my wrongdoing but I seem to get it with both v0.72.1(vanilla from qFlipper) and v0.71.1.
I tried to get confirmation from someone else but got no answer.
